src/com/core/tictactoe/HumanPlayer.java

```diff
@@ -8,12 +8,12 @@ public HumanPlayer(String sign) {
 
     @Override
     public int pickPosition(CommandLineUi commandLineUi, Board board) {
-        int position = commandLineUi.getPositionFromUser(board, this.getSign());
-        return position;
+        String position = commandLineUi.getUserOption(board.getFreePlaces(), this.getPositionPrompt());
```

Comment:
I think it would be nicer if `getUserOption` were a private method as it seems like a detail which the Player doesn't need to know. It would be nice if the commandLineUi exposed a method such as `getMoveChoice(board)`, then `getUserOption` could be called privately inside this method.

Reply:
Thank you Sam, done it :) Also did the same wrapper for getBoardChoice. The commit is here: https://github.com/Kotauror/TicTacToeJava/pull/19/commits/bb0b1c185e90223d2a1afda828ddd1295fa1d49a